core/src/main/clojure/xtdb/metadata.clj

```diff
@@ -281,16 +325,23 @@
     (.put chunks-metadata chunk-idx new-chunk-metadata))
 
   (withMetadata [_ chunk-idx table-name f]
-    (throw (UnsupportedOperationException.))
-    #_
-    (with-single-root buffer-pool (->table-metadata-obj-key chunk-idx table-name)
-      (fn [metadata-root]
-        (.apply f (->table-metadata metadata-root
-                                    (.computeIfAbsent table-metadata-idxs
-                                                      [chunk-idx table-name]
-                                                      (reify Function
-                                                        (apply [_ _]
-                                                          (->table-metadata-idxs metadata-root)))))))))
+    (CompletableFuture/completedFuture
```

Comment:
Should this be somehow become an async call?

Reply:
eventually, sure. might be as straightforward as first doing the `(.getBuffer ...)` call, then calling `util/then-apply` with the rest of the body. you'll also want to ensure the buffer's closed - think a try/finally within the then-apply should suffice this time?

Reply:
An error from the `getBuffer` call would then properly get propagated? Looking at `util/rethrowing-cause` wrapping.